/*
 * This file was automatically generated by EvoSuite
 * Tue Jul 31 22:26:19 GMT 2018
 */

package org.mozilla.javascript.tools;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.lang.MockThrowable;
import org.junit.runner.RunWith;
import org.mozilla.javascript.ErrorReporter;
import org.mozilla.javascript.JavaScriptException;
import org.mozilla.javascript.RhinoException;
import org.mozilla.javascript.WrappedException;
import org.mozilla.javascript.ast.ErrorCollector;
import org.mozilla.javascript.tools.ToolErrorReporter;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false, useJEE = true) 
public class ToolErrorReporter_ESTest extends ToolErrorReporter_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(false);
      toolErrorReporter0.warning(" ", " ", 0, " ", 0);
      assertFalse(toolErrorReporter0.isReportingWarnings());
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      assertTrue(toolErrorReporter0.isReportingWarnings());
      
      toolErrorReporter0.setIsReportingWarnings(false);
      assertFalse(toolErrorReporter0.isReportingWarnings());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      // Undeclared exception!
      try { 
        toolErrorReporter0.runtimeError("+wpIr7+0ZuO/OGP0*t", "?%{g.format2", (-1), "+wpIr7+0ZuO/OGP0*t", (-1));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // -1
         //
         verifyException("org.mozilla.javascript.RhinoException", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(false);
      MockThrowable mockThrowable0 = new MockThrowable();
      WrappedException wrappedException0 = new WrappedException(mockThrowable0);
      toolErrorReporter0.reportException((RhinoException) wrappedException0);
      assertFalse(toolErrorReporter0.isReportingWarnings());
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(false);
      boolean boolean0 = toolErrorReporter0.isReportingWarnings();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      toolErrorReporter0.hasReportedError();
      assertTrue(toolErrorReporter0.isReportingWarnings());
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      // Undeclared exception!
      try { 
        ToolErrorReporter.getMessage("FxI/g5S7} u/<1P{T,d", (Object) "FxI/g5S7} u/<1P{T,d", (Object) toolErrorReporter0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // no message resource found for message property FxI/g5S7} u/<1P{T,d
         //
         verifyException("org.mozilla.javascript.tools.ToolErrorReporter", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      String string0 = ToolErrorReporter.getMessage("msg.format2");
      assertEquals("line {0}: {1}", string0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      ErrorCollector errorCollector0 = new ErrorCollector();
      MockThrowable mockThrowable0 = new MockThrowable();
      WrappedException wrappedException0 = new WrappedException(mockThrowable0);
      // Undeclared exception!
      try { 
        ToolErrorReporter.reportException((ErrorReporter) errorCollector0, (RhinoException) wrappedException0);
        fail("Expecting exception: UnsupportedOperationException");
      
      } catch(UnsupportedOperationException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mozilla.javascript.ast.ErrorCollector", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      // Undeclared exception!
      try { 
        toolErrorReporter0.reportException((RhinoException) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mozilla.javascript.tools.ToolErrorReporter", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      JavaScriptException javaScriptException0 = new JavaScriptException(toolErrorReporter0, "=", 140);
      ToolErrorReporter.reportException((ErrorReporter) toolErrorReporter0, (RhinoException) javaScriptException0);
      assertTrue(toolErrorReporter0.isReportingWarnings());
      assertFalse(toolErrorReporter0.hasReportedError());
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(true);
      toolErrorReporter0.warning("=", "=", (-683), "=", 140);
      assertTrue(toolErrorReporter0.isReportingWarnings());
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      ToolErrorReporter toolErrorReporter0 = new ToolErrorReporter(false);
      toolErrorReporter0.error("r)dW e", (String) null, 115, (String) null, 115);
      assertTrue(toolErrorReporter0.hasReportedError());
  }
}
